validator/coordinator/src/main/java/tech/pegasys/artemis/validator/coordinator/Eth1DataCache.java

```diff
@@ -113,13 +80,34 @@ public Eth1Data get_eth1_vote(BeaconState state) {
     return vote.orElse(defaultVote);
   }
 
-  private NavigableMap<UnsignedLong, Eth1Data> getVotesToConsider() {
-    return eth1ChainCache.subMap(getSpecRangeLowerBound(), true, getSpecRangeUpperBound(), true);
+  private NavigableMap<UnsignedLong, Eth1Data> getVotesToConsider(final UnsignedLong slot) {
+    return eth1ChainCache.subMap(
+        getSpecRangeLowerBound(slot), true, getSpecRangeUpperBound(slot), true);
+  }
+
+  private UnsignedLong calculateCacheDuration() {
+    // Worst case we're in the very last moment of the current slot
+    long cacheDurationSeconds = SECONDS_PER_SLOT;
+
+    // Worst case this slot is at the very end of the current voting period
+    cacheDurationSeconds += (EPOCHS_PER_ETH1_VOTING_PERIOD * SLOTS_PER_EPOCH * SECONDS_PER_SLOT);
+
+    // We need 2 * ETH1_FOLLOW_DISTANCE prior to that
+    cacheDurationSeconds +=
```

Comment:
I think this only needs to be 1 * `ETH1_FOLLOW_DISTANCE` prior, because the latest block events coming from Eth1DataManager already have time, t = (`t - (Eth1 Follow Distance Time) + Request Buffer))`

Reply:
Good spot, fixed.